net/tstun/mtu_test.go

```diff
@@ -4,25 +4,81 @@ package tstun
 
 import (
 	"os"
+	"strconv"
 	"testing"
 )
 
-func TestDefaultMTU(t *testing.T) {
-	orig := os.Getenv("TS_DEBUG_MTU")
-	defer os.Setenv("TS_DEBUG_MTU", orig)
+func TestDefaultTunMTU(t *testing.T) {
+	// Save and restore the envknobs we will be changing
 
+	// The user can specify the MTU with this envknob.
+	ts_debug_mtu := os.Getenv("TS_DEBUG_MTU")
+	defer os.Setenv("TS_DEBUG_MTU", ts_debug_mtu)
 	os.Setenv("TS_DEBUG_MTU", "")
-	if DefaultMTU() != 1280 {
-		t.Errorf("DefaultMTU() = %d, want 1280", DefaultMTU())
+
+	// The intention is that PMTUD will be enabled by default someday.
+	ts_debug_pmtud := os.Getenv("TS_DEBUG_ENABLE_PMTUD")
+	defer os.Setenv("TS_DEBUG_ENABLE_PMTUD", ts_debug_pmtud)
+	os.Setenv("TS_DEBUG_ENABLE_PMTUD", "")
+
+	// With no MTU envknobs, we should get the conservative MTU.
+	if DefaultTunMTU() != safeTunMTU {
+		t.Errorf("DefaultTunMTU() = %d, want %d", DefaultTunMTU(), safeTunMTU)
+	}
+
+	// TS_DEBUG_MTU should take precedence over every other setting.
+	mtu := maxTunMTU - 1
+	os.Setenv("TS_DEBUG_MTU", strconv.Itoa(int(mtu)))
```

Comment:
in general you want `envknob.Setenv` because it handles registered (the more efficient) envknobs, which you might want to switch to later if the envknob lookups are in your hot paths